boot/bootutil/src/bootutil_misc.c

```diff
@@ -248,6 +248,25 @@ boot_enc_key_off(const struct flash_area *fap, uint8_t slot)
 }
 #endif
 
+bool bootutil_buffer_is_erased(const struct flash_area *area,
+                               const void *buffer, size_t len)
+{
+    size_t i;
+    uint8_t *u8b;
+
+    if (buffer == NULL || len == 0) {
+        return false;
+    }
+
+    for (i = 0, u8b = (uint8_t *)buffer; i < len; i++) {
+        if (u8b[i] != flash_area_erased_val(area)) {
```

Comment:
Might be worth caching the return value of `flash_area_erased_val` on entry.  Some implementations might actually require a function in each compare.

Reply:
Done.